### ISSUE TITLE
Image Picker Gallery Upload not actually displaying images.

### ISSUE THREAD

**Issue by matt-fackler** [expo/expo#21603]
### Summary

PLEASE NOTE: I am running all my tests on Expo Go in Android. 
I believe the gallery upload feature of image picker is broken.  I am using useState() to change a placeholder image on screen to one I upload from gallery.  When I try to upload from the gallery the image will not appear.  I expected my image to change to the one I selected from gallery.  Console logs says that the first time I attempt to set the image value, its uri is simply the number 20.  The second time the uri seems correct but the image simply does not appear.  Here is the code I am using.  Please note I am using setTimeout() to make sure the setSelectedImage() function wasn't just delaying.

`const ImageSelect = () => {
    const [selectedImage, setSelectedImage] = useState(require('../includes/img/addimage.png'));

    const uploadImage = async (option) => {
        if (option == 1){
            let result = await ImagePicker.launchImageLibraryAsync({
                allowsEditing: true,
                quality: 1,
            });

            if (!result.canceled){
                console.log(result.assets[0].uri);
                setSelectedImage(result.assets[0].uri);
                setTimeout(() => {
                    console.log(selectedImage);
                }, 10000) 
            }
        } else {
            let result = await ImagePicker.launchCameraAsync({
                allowsEditing: true,
                quality: 1,
            });

            if (!result.canceled){
                console.log(result.assets[0].uri);
                setSelectedImage(result.assets[0].uri);
                
                console.log(selectedImage);
            }
        }
    };

    return (
            <View>
                <Image style={imageSelectUI.mainImage} source={selectedImage}/>
            <View style={{flexDirection: "row"}}>
                <TouchableOpacity style={imageSelectUI.smallButton} onPress={() => uploadImage(0)}>
                    <Text style={{color: 'white'}}>Take Picture</Text></TouchableOpacity>
                <TouchableOpacity style={imageSelectUI.smallButton} onPress={() => uploadImage(1)}>
                    <Text style={{color:'white'}}>Use Gallery</Text></TouchableOpacity>
            </View>
            </View>
        );

}`

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19044
    Binaries:
      Node: 18.12.1 - C:\Program Files\nodejs\node.EXE
      npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD   
    npmPackages:
      expo: ^47.0.13 => 47.0.13
      react: 18.1.0 => 18.1.0
      react-native: 0.70.5 => 0.70.5
    Expo Workflow: managed

### Minimal reproducible example

Here is an example I pulled right off of expos docs from this link: https://docs.expo.dev/tutorial/image-picker/
You can run it in snack if you select it at the bottom of the page.
This does not do what is intended.  The image does not change, not in the online emulators or expo go.
PLEASE NOTE you will have to provide the images but other than that this should be plug and play.

App.js

import { useState } from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, View } from 'react-native';
import * as ImagePicker from 'expo-image-picker';

import ImageViewer from './components/ImageViewer';

const PlaceholderImage = require('./assets/images/background-image.png');

export default function App() {
  const [selectedImage, setSelectedImage] = useState(null);

  const pickImageAsync = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      allowsEditing: true,
      quality: 1,
    });

    if (!result.canceled) {
      setSelectedImage(result.assets[0].uri);
    } else {
      alert("You did not select any image.");
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <ImageViewer placeholderImageSource={PlaceholderImage} selectedImage={selectedImage} />
      </View>
      <View style={styles.footerContainer}>
        <Button theme="primary" label="Choose a photo" onPress={pickImageAsync} />
        <Button label="Use this photo" />
      </View>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#25292e',
    alignItems: 'center',
  },
  imageContainer: {
    flex:1, 
    paddingTop: 58
  },
  footerContainer: {
    flex: 1 / 3,
    alignItems: 'center',
  },
});

ImageViewer.js

import { StyleSheet, Image } from 'react-native';

export default function ImageViewer({ placeholderImageSource, selectedImage }) {
  const imageSource =
    selectedImage !== null ? { uri: selectedImage } : placeholderImageSource;

  return <Image source={imageSource} style={styles.image} />;
}

const styles = StyleSheet.create({  
  image: {
    width: 320,
    height: 440,
    borderRadius: 18,
  },
});

**Comment by alanjhughes** [comment]
Hi, what version of expo go are you using?

**Comment by matt-fackler** [comment]
> Hi, what version of expo go are you using?

I am using expo go version 2.28.4

**Comment by alanjhughes** [comment]
The issue was fixed [here](https://github.com/expo/expo/pull/21589) but it hasn't been published yet.

**Comment by alanjhughes** [comment]
The issue was fixed [here](https://github.com/expo/expo/pull/21589) but it hasn't been published yet.

**Comment by matt-fackler** [comment]
> The issue was fixed [here](https://github.com/expo/expo/pull/21589) but it hasn't been published yet.

Okay great, when it is published will I have to update expo?  Or just the expo-image-picker package?  What commands would I run for that and when can I tell when these changes have been published?

**Comment by alanjhughes** [comment]
You'll just need to update image picker. No way to know when it's published but to keep an eye on the repo. You could also downgrade your expo go version. If you type 'a' after starting metro it should ask you to upgrade expo go but it'll actually install a previous version.

**Comment by matt-fackler** [comment]
> You'll just need to update image picker. No way to know when it's published but to keep an eye on the repo. You could also downgrade your expo go version. If you type 'a' after starting metro it should ask you to upgrade expo go but it'll actually install a previous version.

I unfortunately don't believe this solves my problem.  The issue you linked is about not setting the right permissions and throwing an exception.  I never actually use the requestMediaLibraryPermissionAsync() method in my code.  I just added it to the code with WriteOnly being false by default and still no dice.  My code actually never throws an exception.  It just doesn't display the image whatsoever.  And the first time I run the launchImageLibraryAsync() function it will totally allow me into the gallery but then the images return path is simply "20" and then when I do it the second time the return path seems correct but the image still doesn't show up.
Am I missing something here with the issue you linked?

**Comment by alanjhughes** [comment]
I should have asked at the start but are you running on android 13?

**Comment by matt-fackler** [comment]
I am on Android 13 but I did actually find a solution.  I upgraded all my packages, ran npm install again and then ran npm audit fix --force and then change a bit of my code.  I changed the selected image to {uri: selectedImage} and to account for the initial require() is useState() I just imported the image at the top as "blankPhoto" and then put the initial state as Image.resolveAssetSource(blankPhoto).uri

This fixed it and the image now displays and can change with no errors or warnings.
